src/renderer/store/project.js

```diff
@@ -173,7 +173,14 @@ const actions = {
 
   CREATE_FILE_DIRECTORY ({ commit, state }, name) {
     const { dirname, type } = state.createCache
+
+    if (type === 'file') {
+      const CHECK_MD_EXTENSION = /.*\.(?:markdown|mdown|mkdn|md|mkd|mdwn|mdtxt|mdtext|text|txt)$/i
```

Comment:
What about people that use RMarkdown or use .pandoc as their file extension? Hard coding these seems like an ugly hack that will never cover all the bases. How about just checking if there is any extension at all?

Reply:
I agree to you, but the problem is, that the sidebar does not allow other extensions in the first place. Have a look at src/common/filesystem/paths.js
```
export const MARKDOWN_EXTENSIONS = [
  'markdown',
  'mdown',
  'mkdn',
  'md',
  'mkd',
  'mdwn',
  'mdtxt',
  'mdtext',
  'text',
  'txt'
]

// ...

/**
 * Returns true if the filename matches one of the markdown extensions.
 *
 * @param {string} filename Path or filename
 */
export const hasMarkdownExtension = filename => {
  if (!filename || typeof filename !== 'string') return false
  return MARKDOWN_EXTENSIONS.some(ext => filename.toLowerCase().endsWith(`.${ext}`))
}
```
If you create for example a `.pandoc` file in the sidebar, it is created but Mark Text does not allow to open it and greys it out. That's the reason why I decided to check for these extensions as well.
Maybe we could add the default `.md` extension if no extension is given at all, and warn the user if he creates a file with a disallowed extension?

Reply:
Okay, I guess it would be acceptable for now to match what the sidebar already does, but both the sidebar and this should probably be expanded to cover more options in the future. Would it be possible to use a single constant for this so it doesn't need to be updated in two places and create possible miss-mataches?